pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,762 @@
+}
+
+func (tc *testContext) verify(t *testing.T, expected result, initialObjects []metav1.Object, result interface{}, status *framework.Status) {
+	t.Helper()
+	assert.Equal(t, expected.returnValue, result)
+	assert.Equal(t, expected.status, status)
+	objects := tc.listAll(t)
+	wantObjects := update(t, initialObjects, expected.changes)
+	for _, add := range expected.added {
+		wantObjects = append(wantObjects, add)
+	}
+	for _, remove := range expected.removed {
+		for i, obj := range wantObjects {
+			// This is a bit relaxed (no GVR comparison, no UID
+			// comparison) to simplify writing the test cases.
+			if obj.GetName() == remove.GetName() && obj.GetNamespace() == remove.GetNamespace() {
+				wantObjects = append(wantObjects[0:i], wantObjects[i+1:]...)
+				break
+			}
+		}
+	}
+	sortObjects(wantObjects)
+	stripObjects(wantObjects)
+	stripObjects(objects)
+	assert.Equal(t, wantObjects, objects)
+}
+
+// setGVK is implemented by metav1.TypeMeta and thus all API objects, in
+// contrast to metav1.Type, which is not (?!) implemented.
+type setGVK interface {
+	SetGroupVersionKind(gvk schema.GroupVersionKind)
+}
+
+// stripObjects removes certain fields (Kind, APIVersion, etc.) which are not
+// important and might not be set.
+func stripObjects(objects []metav1.Object) {
+	for _, obj := range objects {
+		obj.SetResourceVersion("")
+		obj.SetUID("")
+		if objType, ok := obj.(setGVK); ok {
+			objType.SetGroupVersionKind(schema.GroupVersionKind{})
+		}
+	}
+}
+
+func (tc *testContext) listAll(t *testing.T) (objects []metav1.Object) {
+	t.Helper()
+	claims, err := tc.client.ResourceV1alpha1().ResourceClaims("").List(tc.ctx, metav1.ListOptions{})
+	require.NoError(t, err, "list claims")
+	for _, claim := range claims.Items {
+		objects = append(objects, &claim)
+	}
+	schedulings, err := tc.client.ResourceV1alpha1().PodSchedulings("").List(tc.ctx, metav1.ListOptions{})
+	require.NoError(t, err, "list pod scheduling")
+	for _, scheduling := range schedulings.Items {
+		objects = append(objects, &scheduling)
+	}
+
+	sortObjects(objects)
+	return
+}
+
+func sortObjects(objects []metav1.Object) {
+	sort.Slice(objects, func(i, j int) bool {
+		if objects[i].GetNamespace() < objects[j].GetNamespace() {
+			return true
+		}
+		return objects[i].GetName() < objects[j].GetName()
+	})
+}
+
+type namespacedName struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+type object struct {
+	Metadata namespacedName `json:"metadata"`
+}
+
+func update[T metav1.Object](t *testing.T, objects []T, updates []any) []T {
+	var updated []T
+
+	// Convert all updates from apply configuration to the corresponding
+	// JSON. By decoding the JSON into namespacedName we can figure out
+	// what the update is for.
+	jsonUpdates := make(map[namespacedName][]byte)
+	for _, update := range updates {
+		data, err := json.Marshal(update)
+		require.NoError(t, err)
+		var o object
+		require.NoError(t, json.Unmarshal(data, &o))
```

Comment:
you should be able to get the name from metav1.Object directly without unmarshaling